src/format/OPUXReader.cpp

```diff
@@ -0,0 +1,287 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "OPUXReader.h"
+
+#include "core/Database.h"
+#include "core/Entry.h"
+#include "core/Group.h"
+#include "core/Metadata.h"
+#include "core/Totp.h"
+
+#include <QFileInfo>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QScopedPointer>
+#include <QUrl>
+
+#include <minizip/unzip.h>
+
+namespace
+{
+    QByteArray extractFile(unzFile uf, QString filename)
+    {
+        if (unzLocateFile(uf, filename.toLatin1(), 2) != UNZ_OK) {
+            qWarning("Failed to extract 1PUX document: %s", qPrintable(filename));
+            return {};
+        }
+
+        // Read export.data into memory
+        int bytes, bytesRead = 0;
+        QByteArray data;
+        unzOpenCurrentFile(uf);
+        do {
+            data.resize(data.size() + 8192);
+            bytes = unzReadCurrentFile(uf, data.data() + bytesRead, 8192);
+            if (bytes > 0) {
+                bytesRead += bytes;
+            }
+        } while (bytes > 0);
+        unzCloseCurrentFile(uf);
+        data.truncate(bytesRead);
+
+        return data;
+    }
+
+    Entry* readItem(const QJsonObject& item, unzFile uf = nullptr)
+    {
+        auto itemMap = item.toVariantMap();
+        auto overviewMap = itemMap.value("overview").toMap();
+        auto detailsMap = itemMap.value("details").toMap();
+
+        // Create entry and assign basic values
+        QScopedPointer<Entry> entry(new Entry());
+        entry->setUuid(QUuid::createUuid());
+        entry->setTitle(overviewMap.value("title").toString());
+        entry->setUrl(overviewMap.value("url").toString());
+        if (overviewMap.contains("urls")) {
+            int i = 1;
+            for (auto urlRaw : overviewMap.value("urls").toList()) {
+                auto urlMap = urlRaw.toMap();
+                auto url = urlMap.value("url").toString();
+                if (entry->url() != url) {
+                    entry->attributes()->set(QString("KP2A_URL_%1").arg(i), url);
+                    ++i;
+                }
+            }
+        }
+        if (overviewMap.contains("tags")) {
+            entry->setTags(overviewMap.value("tags").toStringList().join(","));
+        }
+        if (itemMap.value("favIndex").toString() == "1") {
+            entry->addTag(QObject::tr("Favorite", "Tag for favorite entries"));
+        }
+        if (itemMap.value("state").toString() == "archived") {
+            entry->addTag(QObject::tr("Archived", "Tag for archived entries"));
+        }
+
+        // Parse the details map by setting the username, password, and notes first
+        auto loginFields = detailsMap.value("loginFields").toList();
+        for (const auto& field : loginFields) {
```

Comment:
```suggestion
        const auto loginFields = detailsMap.value("loginFields").toList();
        for (const auto& field : loginFields) {
```

Or use it as rvalue.